app/src/main/java/com/android/unio/model/search/SearchRepository.kt

```diff
@@ -0,0 +1,211 @@
+package com.android.unio.model.search
+
+import android.content.Context
+import android.util.Log
+import androidx.appsearch.app.AppSearchSession
+import androidx.appsearch.app.PutDocumentsRequest
+import androidx.appsearch.app.RemoveByDocumentIdRequest
+import androidx.appsearch.app.SearchSpec
+import androidx.appsearch.app.SetSchemaRequest
+import androidx.appsearch.localstorage.LocalStorage
+import com.android.unio.model.association.Association
+import com.android.unio.model.association.AssociationDocument
+import com.android.unio.model.association.AssociationRepository
+import com.android.unio.model.association.toAssociationDocument
+import com.android.unio.model.event.Event
+import com.android.unio.model.event.EventDocument
+import com.android.unio.model.event.EventRepository
+import com.android.unio.model.event.toEventDocument
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlin.coroutines.suspendCoroutine
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.guava.await
+import kotlinx.coroutines.withContext
+
+/** Repository for searching associations and events */
+class SearchRepository(
+    private val appContext: Context,
+    private val associationRepository: AssociationRepository,
+    private val eventRepository: EventRepository
+) {
+  // Should be private, but I need it public for tests
+  var session: AppSearchSession? = null
```

Comment:
Tests should focus only on the input/output of classes, **not** the internal behavior. This means that either this property should be private and not tested, or public and part of the SearchRepository's API. Testing internal behavior is bad practice as it might be subject to change at any moment.